src/worker.cpp

```diff
@@ -466,31 +469,70 @@ void Worker::setDeviceTreeAndJson()
                                         ec);
         if (ec)
         {
-            throw std::runtime_error(
-                "create_symlink system call failed with error" + ec.message());
+            if (ec.value() == EEXIST)
+            {
+                ec.clear();
+                logging::logMessage(
+                    "Sym link already exists, file [" +
+                    std::string(INVENTORY_JSON_SYM_LINK) +
+                    "] deleting and creating sym link again, target file: " +
+                    systemJson);
+
+                if (std::filesystem::remove(INVENTORY_JSON_SYM_LINK, ec))
+                {
+                    std::filesystem::create_symlink(
+                        systemJson, INVENTORY_JSON_SYM_LINK, ec);
+                    if (ec)
+                    {
+                        throw std::runtime_error(
+                            "create_symlink system call failed when deleting existing link and recreating it again, error: " +
+                            ec.message());
+                    }
+                }
+                else
+                {
+                    logging::logMessage(
+                        "remove system call failed for file[" +
+                        std::string(INVENTORY_JSON_SYM_LINK) + "], error[" +
+                        ec.message() + "], continuing with existing sym link.");
+                }
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "create_symlink system call failed with error: " +
+                    ec.message());
+            }
         }
 
         // re-parse the JSON once appropriate JSON has been selected.
         try
         {
-            m_parsedJson = nlohmann::json::parse(INVENTORY_JSON_SYM_LINK);
+            m_parsedJson = jsonUtility::getParsedJson(INVENTORY_JSON_SYM_LINK);
         }
         catch (const nlohmann::json::parse_error& ex)
         {
             throw(JsonException("Json parsing failed", systemJson));
         }
     }
 
-    auto devTreeFromJson = m_parsedJson["devTree"];
-    if (devTreeFromJson.empty())
+    std::string devTreeFromJson;
+    if (m_parsedJson.contains("devTree"))
     {
-        throw JsonException("Mandatory value for device tree missing from JSON",
-                            INVENTORY_JSON_SYM_LINK);
+        devTreeFromJson = m_parsedJson["devTree"];
+
+        if (devTreeFromJson.empty())
+        {
+            throw JsonException(
+                "Mandatory value for device tree missing from JSON",
+                INVENTORY_JSON_SYM_LINK);
+        }
     }
 
     auto fitConfigVal = readFitConfigValue();
 
-    if (fitConfigVal.find(devTreeFromJson) != std::string::npos)
+    if (devTreeFromJson.empty() ||
+        fitConfigVal.compare(devTreeFromJson) == constants::STR_CMP_SUCCESS)
     { // fitconfig is updated and correct JSON is set.
```

Comment:
Since you have changed the condition, this comment is not valid now. Even if fitconfig is not set with correct JSON it will enter the if condition.
Please add a detailed comment here.
"Skipping setting device tree as either devtree info is missing from Json or it is rightly set."